astropy/table/table.py

```diff
@@ -785,35 +785,55 @@ def _new_from_slice(self, slice_):
         table.meta.clear()
         table.meta.update(deepcopy(self.meta))
         table.primary_key = self.primary_key
-        cols = self.columns.values()
 
         newcols = []
-        for col in cols:
-            col.info._copy_indices = self._copy_indices
+        for col in self.columns.values():
             newcol = col[slice_]
-            if col.info.indices:
+
+            # Note in line below, use direct attribute access to col.indices for Column
+            # instances instead of the generic col.info.indices.  This saves about 4 usec
+            # per column.
+            if (col if isinstance(col, Column) else col.info).indices:
+                # TODO : as far as I can tell the only purpose of setting _copy_indices
+                # here is to communicate that to the initial test in `slice_indices`.
+                # Why isn't that just sent as an arg to the function?
+                col.info._copy_indices = self._copy_indices
                 newcol = col.info.slice_indices(newcol, slice_, len(col))
+
+                # Don't understand why this is forcing a value on the original column.
```

Comment:
Agreed, this looks like a big. If anything, it should keep whatever the old value was and restore it here.

Probably best for another PR, but maybe worth raising an issue?

Reply:
I opened #8501.